cli/js/files.ts

```diff
@@ -231,5 +280,75 @@ export type OpenMode =
  * @internal
  */
 export function create(filename: string): Promise<File> {
-  return open(filename, "w+");
+  return open(filename, {
+    read: true,
+    create: true,
+    write: true,
+    truncate: true
+  });
+}
+
+/** Check if OpenMode is set to something that is valid.
+ *  @returns Tuple representing if openMode is valid and error message if it's not
+ *  @internal
+ */
+function checkOpenCapability(mode: OpenCapability): [boolean, string] {
+  const allOptionsAreFalse =
+    Object.values(mode).filter(val => val == true).length === 0;
+  const truncateOptionWithoutWriteAccess = mode.truncate && !mode.write;
+  const createOrCreateNewWithoutWriteOrAppend =
+    (mode.create || mode.createNew) && !(mode.write || mode.append);
+  if (allOptionsAreFalse)
+    return [false, "OpenMode require at least one option to be true"];
+  if (truncateOptionWithoutWriteAccess)
+    return [false, "Truncate option require write accesss"];
+  if (createOrCreateNewWithoutWriteOrAppend)
+    return [false, "Create or create new requires write or append access"];
+  return [true, ""];
+}
+
+const openModMap = {
```

Comment:
```suggestion
const openModeMap = {
```
Is this a missing on `e` or just shorthand?

Reply:
missing, thank you